packages/cli/src/util/integration/types.ts

```diff
@@ -32,12 +32,27 @@ export interface MetadataSchema {
   required?: string[];
 }
 
+export type IntegrationProductProtocolBase = {
+  status: 'enabled' | 'disabled';
+};
+
+export type StorageIntegrationProtocol = IntegrationProductProtocolBase & {
+  repl?: {
+    enabled: boolean;
+    supportsReadOnlyMode: boolean;
+    welcomeMessage?: string;
+  };
+};
+
 export interface IntegrationProduct {
   id: string;
   slug: string;
   name: string;
   shortDescription: string;
-  type: 'storage' | string;
+  type?: 'storage' | string;
+  protocols?: {
+    storage?: StorageIntegrationProtocol;
+  };
   metadataSchema: MetadataSchema;
 }
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏use union types for stronger type checking

When the overall true type varies when the value of a property, I like to use a union type for stronger type checking. This will prevent the case where both `storage` and `protocols` are missing or both are populated.

Example:

```ts
type IntegrationProductBase = {
  id: string;
  slug: string;
  name: string;
  shortDescription: string;
  metadataSchema: MetadataSchema;
}
type OldIntegrationProduct = IntegrationProductBase & {
  // `type` no longer optional
  type: 'storage' | string;
}

type NewIntegrationProduct = IntegrationProductBase & {
  // `protocols` no longer optional
  protocols: {
    storage?: StorageIntegrationProtocol;
  };
}

type IntegrationProduct = OldIntegrationProduct | NewIntegrationProduct;

let a: IntegrationProduct = {
  id: 'a',
  slug: 'product-a',
  name: 'Product A',
  shortDescription: 'product called "a"',
  metadataSchema: {},

  type: 'storage',
  
}

let b: IntegrationProduct = {
  id: 'b',
  slug: 'product-b',
  name: 'Product B',
  shortDescription: 'product called "B"',
  metadataSchema: {},

  protocols: {
    storage: {}
  },
}
```

Reply:
In our world, both fields can exist, so it's less a "one or the other" case. We've introduced this new concept of protocols, which the old integrations also use, and the protocols just-so-happen to do what "type" did as well, but better (product being multiple categories and categories having some bespoke concepts). Eventually we'll deprecate the type field and remove it entirely, but for now we have integrations where both values exist.

I'll keep this in mind for future situations, though 🙂

Reply:
Ah, my reading from the PR description was that they were exclusive.

Do the values need to exist in responses from the API? If not, you could restrict those values at the API serialization layer such that types like this suggestion could be used in clients.